packages/next/src/build/webpack-config.ts

```diff
@@ -819,6 +819,7 @@ export default async function getBaseWebpackConfig(
   const shouldIncludeExternalDirs =
     config.experimental.externalDir || !!config.transpilePackages
 
+  const pageExtensionsRegex = new RegExp(`(${pageExtensions.join('|')})$`)
```

Comment:
Are `pageExtensions` prefixed with a `.`? If so then we need to escape that. Otherwise it matches any character. If not, then we should use 
```suggestion
  const pageExtensionsRegex = new RegExp(`\.(${pageExtensions.join('|')})$`)
```
instead so that we don't match extensions that just happen to share a suffix.